paasta_tools/tron_tools.py

```diff
@@ -395,13 +395,22 @@ def _get_tron_namespaces_from_tron_dir(cluster, soa_dir):
     ]
     return namespaces
 
+class ConflictingNamespacesError(RuntimeError):
+    pass
 
 def get_tron_namespaces_for_cluster(cluster=None, soa_dir=DEFAULT_SOA_DIR):
     """Get all the namespaces that are configured in a particular Tron cluster."""
     if not cluster:
         cluster = load_tron_config().get_cluster_name()
 
-    namespaces1 = _get_tron_namespaces_from_service_dir(cluster, soa_dir)
-    namespaces2 = _get_tron_namespaces_from_tron_dir(cluster, soa_dir)
-    namespaces = list(set(namespaces1 + namespaces2))
+    namespaces1 = set(_get_tron_namespaces_from_service_dir(cluster, soa_dir))
+    namespaces2 = set(_get_tron_namespaces_from_tron_dir(cluster, soa_dir))
+
+    if namespaces1.intersection(namespaces2):
```

Comment:
I'm not sure on the context, but I kind of think this error checking logic belongs in the caller?

Reply:
result of this function is list of unique namespaces, caller can't check if there were any duplicates.

this function knows that namespaces can come from multiple sources - i think we should raise an error as fast and close to the source as possible.